core/state/statedb.go

```diff
@@ -486,7 +486,14 @@ func (db *StateDB) ForEachStorage(addr common.Address, cb func(key, value common
 			cb(key, value)
 			continue
 		}
-		cb(key, common.BytesToHash(it.Value))
+
+		if len(it.Value) > 0 {
+			_, content, _, err := rlp.Split(it.Value)
+			if err != nil {
+				continue
```

Comment:
This seems like a very bad idea. When we iterate a trie, no error should ever happen. Here's not only are we expecting an error, but also silently discarding it. This would end up haunting us a lot. What's happening here?